### ISSUE TITLE
A1111 Encoding Falled

### ISSUE THREAD

**Issue by Lookplanes** [asagi4/comfyui-prompt-control#100]
Hello, I think it is strange that the A1111 advanced encodings has some problems.
The A1111 style encoding mode may results the following error:

![Image](https://github.com/user-attachments/assets/1374763b-4452-4705-964a-7447bbec998c)

The prompt in the screenshot:
`STYLE(A1111)
1girl, 
masterpiece, best quality, newest, masterpiece, best quality, very aesthetic, highres, highres, absurdres`

(I only use a very short positive prompt in the screenshot. Exactly, if i use much longer prompt or using BREAK, the second tensor size  keeps half of the size of the first one (like "The expanded size of the tensor (308) must match the existing size (154) ……" as another example)

For other encoding method, they work well (At least no error happens).

The detailed error message is:
`# ComfyUI Error Report
## Error Details
- **Node ID:** 540
- **Node Type:** PCLazyTextEncode
- **Exception Type:** RuntimeError
- **Exception Message:** The expanded size of the tensor (154) must match the existing size (77) at non-singleton dimension 1.  Target sizes: [1, 154, 1280].  Tensor sizes: [1, 77, 1]
## Stack Trace
```
  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\execution.py", line 327, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\execution.py", line 202, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\execution.py", line 174, in _map_node_over_list
    process_inputs(input_dict, i)

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\execution.py", line 163, in process_inputs
    results.append(getattr(obj, func)(**inputs))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-prompt-control\prompt_control\nodes_base.py", line 43, in apply
    return PCTextEncodeWithRange.apply(self, clip, text, 0.0, 1.0)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-prompt-control\prompt_control\nodes_base.py", line 27, in apply
    return (encode_prompt(clip, text, start, end, defaults, masks),)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-prompt-control\prompt_control\prompts.py", line 474, in encode_prompt
    x = encode_prompt_segment(clip, prompt, settings, style, normalization)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-prompt-control\prompt_control\prompts.py", line 203, in encode_prompt_segment
    return clip.encode_from_tokens_scheduled(tokens, add_dict=settings)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\comfy\sd.py", line 149, in encode_from_tokens_scheduled
    pooled_dict = self.encode_from_tokens(tokens, return_pooled=return_pooled, return_dict=True)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\comfy\sd.py", line 211, in encode_from_tokens
    o = self.cond_stage_model.encode_token_weights(tokens)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\comfy\sdxl_clip.py", line 60, in encode_token_weights
    g_out, g_pooled = self.clip_g.encode_token_weights(token_weight_pairs_g)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-prompt-control\prompt_control\prompts.py", line 234, in encode
    r = advanced_encode_from_tokens(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-prompt-control\prompt_control\adv_encode.py", line 212, in advanced_encode_from_tokens
    weighted_emb = base_emb * weights_like(weights, base_emb)  # from_zero
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\comfyui\ComfyUI_windows_portable_nvidia_new\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-prompt-control\prompt_control\adv_encode.py", line 22, in weights_like
    return torch.tensor(weights, dtype=emb.dtype, device=emb.device).reshape(1, -1, 1).expand(emb.shape)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^`

Sorry for any inconvience that I caused. Looking forward to your reply.

**Comment by asagi4** [comment]
Do you have a workflow to reproduce this? Because A1111 encoding works fine for me; you might have some other nodes interfering.

**Comment by Lookplanes** [comment]
I'm closing this issue as the problem seems to be resolved. It appears there was an issue with my ComfyUI installation after updating to the latest version. Reinstalling ComfyUI (still the latest version) fixed the problem.

Not sure exactly what went wrong during the update, but a fresh install cleared it up.

Thanks for your time!
